source/server/filter_chain_manager_impl.cc

```diff
@@ -23,6 +23,104 @@ Network::Address::InstanceConstSharedPtr fakeAddress() {
 
 } // namespace
 
+FilterChainFactoryContextImpl::FilterChainFactoryContextImpl(
+    Configuration::FactoryContext& parent_context, uint64_t filter_chain_tag)
+    : parent_context_(parent_context), filter_chain_tag_(filter_chain_tag) {}
+
+// FilterChainFactoryContext
+uint64_t FilterChainFactoryContextImpl::filterChainTag() const { return filter_chain_tag_; }
+
+// DrainDecision
+bool FilterChainFactoryContextImpl::drainClose() const {
+  // TODO(lambdai): will provide individual value for each filter chain context.
+  return parent_context_.drainDecision().drainClose();
+}
+
+Network::DrainDecision& FilterChainFactoryContextImpl::drainDecision() { return *this; }
+
+// TODO(lambdai): init manager will be provided for each filter chain update.
+Init::Manager& FilterChainFactoryContextImpl::initManager() {
+  return parent_context_.initManager();
+}
+
+// Delegate to parent context
+ThreadLocal::SlotAllocator& FilterChainFactoryContextImpl::threadLocal() {
+  return parent_context_.threadLocal();
+}
+
+const envoy::api::v2::core::Metadata& FilterChainFactoryContextImpl::listenerMetadata() const {
+  return parent_context_.listenerMetadata();
+};
+envoy::api::v2::core::TrafficDirection FilterChainFactoryContextImpl::direction() const {
```

Comment:
nit: consistent newline between method bodies, here and elsewhere.

Reply:
will fix

Reply:
Method definition has extra newline. Fixed